packages/data-flow/src/orchestrator.ts

```diff
@@ -0,0 +1,235 @@
+import { IIndexerClient } from "@grants-stack-indexer/indexer-client";
+import {
+    existsHandler,
+    UnsupportedEventException,
+    UnsupportedStrategy,
+} from "@grants-stack-indexer/processors";
+import {
+    Address,
+    AnyEvent,
+    ChainId,
+    ContractName,
+    Hex,
+    isAlloEvent,
+    isStrategyEvent,
+    ProcessorEvent,
+    StrategyEvent,
+    stringify,
+} from "@grants-stack-indexer/shared";
+
+import { EventsFetcher } from "./eventsFetcher.js";
+import { EventsProcessor } from "./eventsProcessor.js";
+import { InvalidEvent } from "./exceptions/index.js";
+import { IEventsRegistry } from "./interfaces/eventsRegistry.interface.js";
+import { IEventsFetcher } from "./interfaces/index.js";
+import { IStrategyRegistry } from "./interfaces/strategyRegistry.interface.js";
+import { CoreDependencies, DataLoader, delay, IQueue, iStrategyAbi, Queue } from "./internal.js";
+
+/**
+ * The Orchestrator is the central coordinator of the data flow system, managing the interaction between
+ * three main components:
+ *
+ * 1. Events Fetcher: Retrieves blockchain events from the indexer service
+ * 2. Events Processor: Processes and transforms raw events into domain events
+ * 3. Data Loader: Persists the processed events into the database
+ *
+ * The Orchestrator implements a continuous processing loop that:
+ *
+ * 1. Fetches batches of events from the indexer and stores them in an internal queue
+ * 2. Processes each event from the queue:
+ *    - For strategy events and PoolCreated from Allo contract, enhances them with strategyId
+ *    - Forwards the event to the Events Processor which is in charge of delagating the processing of the event to the correct handler
+ *    - Discards events for unsupported strategies or events
+ * 3. Loads the processed events into the database via the Data Loader
+ *
+ * The Orchestrator provides fault tolerance and performance optimization through:
+ * - Configurable batch sizes for event fetching
+ * - Delayed processing to prevent overwhelming the system
+ * - Error handling and logging for various failure scenarios
+ * - Registry tracking of supported/unsupported strategies and events
+ *
+ * TODO: Enhance the error handling/retries, logging and observability
+ * TODO: Handle unhandled strategies appropriately
+ */
+export class Orchestrator {
+    private readonly eventsQueue: IQueue<ProcessorEvent<ContractName, AnyEvent>>;
```

Comment:
we won't probably but since we follow best practices, seemed nice here too. maybe we implement a super mega optimized new version of the Queue class idk